src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx

```diff
@@ -206,7 +206,7 @@ export class CreateDataSourceForm extends React.Component<
   };
 
   validateRegion = () => {
-    const isValid = !!this.state.auth.credentials.region?.trim().length;
+    const isValid = !!this.state.auth.credentials.region?.toString().trim().length;
```

Comment:
The same here, if we expect `region` to be a non-empty string, I'd do this validation:

```typescript
const isValid = typeof this.state.auth.credentials.region === 'string' && this.state.auth.credentials.region.trim().length > 0;

```